test/errhandling/stmtVsExpr.chpl

```diff
@@ -0,0 +1,46 @@
+var x, y: int;
+class C { }
+var myC = new unmanaged C();
+
+try! x = y;
+try! x <=> y;
+try! x += y;
+try! if x >= y then writeln("Hi");
+try! for i in 0..1 do writeln(i);
+try! select x { when -100 do writeln("Yo"); otherwise writeln("Bye"); }
+try! foo();
+sync {
+  try! begin foo();
+}
+try! sync { begin foo(); }
+try! cobegin { ; foo(); }
+try! delete myC;
+try! local { foo(); }
+try! on Locales[numLocales-1] do foo();
+try! serial { foo(); }
```

Comment:
This might be irritating, but could we add a test that ensures the halting property of `try!` is working for these new forms?

Reply:
I don't particularly want to (or have time to) make a test per statement type and would prefer to rely on orthogonality (`try!` generally halts correctly and this change just permits it to be syntactically applied to more things) if that's acceptable to you.  I would be willing to add a test of one or two of the forms if you have a favorite you'd like to see tested.

Reply:
Nah, I felt obligated to say it but realize it's a PITA. You could just try it locally and that would be enough for me.

Reply:
OK, thanks.  I think I will add one test because I'm noting that different code paths are used in the new parser for block_stmts vs. executable_stmts, so that feels like an appropriate thing to do from the perspective of whitebox testing.

Reply:
Added a new test for this and my original now halts on a `try! throw...` as well (so we've got two cases now).